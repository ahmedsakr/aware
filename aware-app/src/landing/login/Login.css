#login {
    background-color: #232f3e;
    border-radius: 7px;
    padding: 20px;
}

#login > h2, #login > h4 {
    color: #c1c7c9;
    text-align: center;    
}

#login-username, #login-password {
    margin-top: 15px;
}

#login-register {
    margin-top: 10px;
}

#forgot-password {
    margin-top: 10px;
    float: right;
}

#remember-me {
    margin-top: 10px;
}