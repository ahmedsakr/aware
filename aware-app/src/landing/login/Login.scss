/*
 * Loing styling
 *
 * Defines layout fundamentals for the login page.
 */


$background_color: #232f3e;
$font_color: #c1c7c9;

@mixin top_margin($value) {
    margin-top: $value
}

#login {
    background-color: $background_color;
    border-radius: 7px;
    padding: 20px;

    > {

        h2,
        h4 {
            color: $font_color;
            text-align: center;
        }
    }
}

#login-username,
#login-password {
    @include top_margin(15px);
}

#login-register {
    @include top_margin(10px);
}

#forgot-password {
    float: right;
}

#checkbox-forgot {
    @include top_margin(10px);
}