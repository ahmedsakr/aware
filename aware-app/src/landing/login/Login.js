import React, { Component } from 'react';
import "./Login.css";

import Cookies from 'universal-cookie';
const cookies = new Cookies();

class Login extends Component {
    constructor() {
        super()
        this.state = {
            username: cookies.get('aware-user'),
            rememberMe: false
        }
        this.handleUsernameChange = this.handleUsernameChange.bind(this);
        this.handleCheckboxChange = this.handleCheckboxChange.bind(this);
    }

    render() {
        return(
            <div id="login">
                <h2 id="welcome-message">Welcome back to Aware</h2>
                <h4>Login now to gain access</h4>
                <form>
                    <div class="container">
                        <label id="login-username">Username</label>
                        <input class="landing-textfield" type="text" value={this.state.username} onChange={this.handleUsernameChange}></input>

                        <label id="login-password">Password</label>

                        <input class="landing-textfield" type="password"></input>

                        <label id="remember-me">
                            <input type="checkbox" onChange={this.handleCheckboxChange}/> 
                            Remember me
                        </label>
                        <label id="forgot-password"><span>Forgot password?</span></label>

                        <button type="submit" onClick={() => {this.props.setUsername(this.state.username); this.props.loadMessenger(this.state.rememberMe)}}>Login</button>

                        <label id="login-register">Don't have an account? <span onClick={() => this.props.switch()}>register now!</span></label>
                    </div>
                </form>
            </div>
        );
    }

    handleUsernameChange = (event) => {
        this.setState({username: event.target.value});
    }

    handleCheckboxChange = (event) => {
        this.setState({ rememberMe: event.target.checked});
    }
}

export default Login;