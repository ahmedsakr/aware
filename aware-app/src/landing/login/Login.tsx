import React from 'react';
import verification from '../../shared/verification/user';
import "./Login.css";

import Cookies from 'universal-cookie';
const cookies: Cookies = new Cookies();

type LoginProps = {
    setUsername: (username: string) => void,
    switch: () => void,
    socket: SocketIOClient.Socket 
};

type LoginState = {
    username: string | undefined,
    password: string,
    rememberMe: boolean
};

class Login extends React.Component<LoginProps, LoginState> {
    constructor(props: LoginProps) {
        super(props);

        this.state = {
            username: cookies.get('aware-user'),
            password: "",
            rememberMe: cookies.get('aware-user') !== undefined
        }

        this.handleChange = this.handleChange.bind(this);
    }

    login(): void {
        if (!verification.verifyUsername(this.state.username)) {
            alert("Please provide a username between 3 and 32 characters.");
            return;
        }

        if (!verification.verifyPassword(this.state.password)) {
            alert("Please provide a password between 8 and 128 characters.");
            return;
        }

        if (this.state.rememberMe) {
            cookies.set('aware-user', this.state.username, {path: '/'});
        } else {
            cookies.remove('aware-user', {path: '/'})
        }

        this.props.setUsername(this.state.username as string);
        this.props.socket.emit('login', this.state.username, this.state.password);
    }

    handleChange(event: React.ChangeEvent<HTMLInputElement>) {
        let value: string | boolean =
            event.type === 'checkbox' ? event.target.checked : event.target.value;

        this.setState({
            [event.target.name]: value
        } as any); 
    }

    render() {
        return(
            <div id="login">
                <h2 id="welcome-message">Welcome back to Aware</h2>
                <h4>Login now to gain access</h4>
                <div className="container">
                    <label id="login-username">Username</label>
                    <input
                        className="landing-textfield"
                        name="username"
                        type="text"
                        value={this.state.username}
                        onChange={this.handleChange} />

                    <label id="login-password">Password</label>
                    <input
                        className="landing-textfield"
                        name="password"
                        type="password"
                        value={this.state.password}
                        onChange={this.handleChange} />

                    <div id="checkbox-forgot">
                        <label>
                            <input
                                type="checkbox"
                                name="rememberMe"
                                onChange={this.handleChange}
                                checked={this.state.rememberMe} />
                            &nbsp; Remember me
                            </label>
                        <span id="forgot-password">Forgot password?</span>
                    </div>

                    <button type="submit" onClick={() => { this.login() }}>Login</button>

                    <label id="login-register">Don't have an account? <span onClick={() => this.props.switch()}>register now!</span></label>
                </div>  
            </div>
        );
    }
}

export default Login;