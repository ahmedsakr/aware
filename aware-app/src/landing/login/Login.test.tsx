import React from 'react';
import { shallow } from 'enzyme';

import Login from './Login';
import io from 'socket.io-client';

jest.mock('socket.io-client');

describe('Login', () => {
    it('should render correctly', () => {

        // Stub out the function for now, but we probably need to
        // implement better tests in the future.
        let setUsername = (username: string) => { }
        let switchStub = () => { }
        let channel = io();

        const component = shallow(
            <Login
                setUsername={setUsername}
                switch={switchStub}
                socket={channel} />
        );

        expect(component).toMatchSnapshot();
    });
});