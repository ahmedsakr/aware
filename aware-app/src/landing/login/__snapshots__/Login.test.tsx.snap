version https://git-lfs.github.com/spec/v1
oid sha256:2099f8171ec570deac35714993d9a81c7739ae530584bf30a6b52a3c5b5af26b
size 1302
